src/authenticator.ts

```diff
@@ -13,28 +8,40 @@ export interface AuthenticateCallback<User> {
 /**
  * Extra options for the authenticator.
  */
-export interface AuthenticatorOptions {
-  sessionKey?: AuthenticateOptions["sessionKey"];
-  sessionErrorKey?: AuthenticateOptions["sessionErrorKey"];
-  sessionStrategyKey?: AuthenticateOptions["sessionStrategyKey"];
-  throwOnError?: AuthenticateOptions["throwOnError"];
+export interface PublicAuthenticatorOptions<TUser, TFlash, TContext> {
+  sessionStorage: SessionStorage<TUser, TFlash>;
+  sessionKey?: AuthenticateOptions<TContext>["sessionKey"];
+  sessionErrorKey?: AuthenticateOptions<TContext>["sessionErrorKey"];
+  sessionStrategyKey?: AuthenticateOptions<TContext>["sessionStrategyKey"];
+  throwOnError?: AuthenticateOptions<TContext>["throwOnError"];
 }
 
-export class Authenticator<User = unknown> {
+export interface AuthenticatorImplOptions {
+  redirect: (
+    url: string,
+    options?: { headers?: HeadersInit }
+  ) => Response | Promise<Response>;
+  json: <T>(data: T, status?: number) => Response;
+}
```

Comment:
Do you need to provide this from TanStack? At least in Remix they only wrap `new Response()`, in that case they could be implemented right here